src/rpc/ngHandlers/LedgerData.cpp

```diff
@@ -0,0 +1,221 @@
+    }
+    else
+    {
+        if (input.marker && !sharedPtrBackend_->fetchLedgerObject(*(input.marker), lgrInfo.seq, ctx.yield))
+            return Error{RPC::Status{RPC::RippledError::rpcINVALID_PARAMS, "markerDoesNotExist"}};
+    }
+
+    output.ledgerHash = ripple::strHex(lgrInfo.hash);
+    output.ledgerIndex = lgrInfo.seq;
+
+    auto const start = std::chrono::system_clock::now();
+    std::vector<Backend::LedgerObject> results;
+    if (input.diffMarker)
+    {
+        // keep the same logic as previous implementation
+        auto const diff = sharedPtrBackend_->fetchLedgerDiff(*(input.diffMarker), ctx.yield);
+        std::vector<ripple::uint256> keys;
+        for (auto& [key, object] : diff)
+        {
+            if (!object.size())
+                keys.push_back(std::move(key));
+        }
+        auto objs = sharedPtrBackend_->fetchLedgerObjects(keys, lgrInfo.seq, ctx.yield);
+        for (size_t i = 0; i < objs.size(); ++i)
+        {
+            auto& obj = objs[i];
+            if (obj.size())
+                results.push_back({std::move(keys[i]), std::move(obj)});
+        }
+        if (*(input.diffMarker) > lgrInfo.seq)
+            output.diffMarker = *(input.diffMarker) - 1;
+    }
+    else
+    {
+        // limit's limitation is different based on binary or json
+        // framework can not handler the check right now, adjust the value here
+        auto const limit =
+            std::min(input.limit, input.binary ? LedgerDataHandler::LIMITBINARY : LedgerDataHandler::LIMITJSON);
+        auto page = sharedPtrBackend_->fetchLedgerPage(input.marker, lgrInfo.seq, limit, input.outOfOrder, ctx.yield);
+        results = std::move(page.objects);
+        if (page.cursor)
+            output.marker = ripple::strHex(*(page.cursor));
+        else if (input.outOfOrder)
+            output.diffMarker = sharedPtrBackend_->fetchLedgerRange()->maxSequence;
+    }
+    auto const end = std::chrono::system_clock::now();
+
+    auto time = std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
+
+    log_.debug() << "Number of results = " << results.size() << " fetched in " << time << " microseconds";
+    output.states.reserve(results.size());
+    for (auto const& [key, object] : results)
+    {
+        ripple::STLedgerEntry const sle{ripple::SerialIter{object.data(), object.size()}, key};
+        if (input.binary)
+        {
+            boost::json::object entry;
+            entry[JS(data)] = ripple::serializeHex(sle);
+            entry[JS(index)] = ripple::to_string(sle.key());
+            output.states.push_back(std::move(entry));
+        }
+        else
+        {
+            output.states.push_back(RPC::toJson(sle));
+        }
+    }
+    if (input.outOfOrder)
+        output.cacheFull = sharedPtrBackend_->cache().isFull();
+    auto const end2 = std::chrono::system_clock::now();
+
+    time = std::chrono::duration_cast<std::chrono::microseconds>(end2 - end).count();
```

Comment:
no need to try reuse, just create a new const here :) i understand this is old code but let's improve it

Reply:
you can even put it straight into the log line.. no real need for temporaries here